src/PowerShellEditorServices/Workspace/ScriptFile.cs

```diff
@@ -316,13 +316,17 @@ public void ValidatePosition(int line, int column)
 
         /// <summary>
         /// Throws ArgumentOutOfRangeException if the given position is outside
-        /// of the file's buffer extents.
+        /// of the file's buffer extents. If the position is for an insertion (an applied change)
+        /// the index may be 1 past the end of the file, which is just appended.
         /// </summary>
         /// <param name="line">The 1-based line to be validated.</param>
         /// <param name="column">The 1-based column to be validated.</param>
         /// <param name="isInsertion">If true, the position to validate is for an applied change.</param>
         public void ValidatePosition(int line, int column, bool isInsertion)
         {
+            // If new content is being added, VSCode sometimes likes to add it a (FileLines.Count + 1),
```

Comment:
Minor nit, should this be "add it *at* …"?

Reply:
Are these comments ok @rkeithhill? Want to make sure I've put them where you want and saying enough, etc.

Reply:
Yes, they LGTM.